### PR TITLE
922 nginx uwsgi fix

### PR DESCRIPTION
**Pull request by mjstealey**
Updates to uwsgi and nginx
- accommodate up to 4 GB file download (to be increased at a later date)
- use unix socket for uWSGI server
- update nginx config files for unix socket and 4 GB file download and clean up spacing
- update hsctl restart to use docker-compose calls instead of docker call
- update docker-compose.template for unix socket use
- update Dockerfile to use docker hub tagged build ID

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by pkdash** [comment]
@hyi Please review

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM mjstealey/hydroshare_base
+FROM mjstealey/hs_docker_base:1.6.5
```
Comment (ironfroggy) [inline Dockerfile]: :+1:

File: hsctl [hunk 2]
```diff
@@ -60,8 +60,9 @@ restart_hs() {
     echo "*** ${1^^} ***"
     stop_nginx
     preflight_hs
-    docker stop hydroshare_hydroshare_1
-    docker start hydroshare_hydroshare_1
+    docker-compose stop hydroshare
+    docker-compose build hydroshare
```
Comment (hyi) [inline hsctl]: @mjstealey Do we need to run "docker-compose build" for restart_hs()? Is 'docker-compose up' followed by 'docker-compose stop' sufficient?

**Comment by hyi** [comment]
@mjstealey I only have one minor comment. Otherwise, +1 from me.

File: hsctl (hunk 2 above)
Comment (mjstealey) [inline hsctl]: @hyi - Ah yes, good catch. That line was initially a `docker-compose start hydroshare` call but changed it to `up` for the very reason that up builds and runs. Will remove the `build` call and update the PR.

**Pull request merged**